politeiawww/user/user.go

```diff
@@ -373,12 +380,51 @@ type Plugin struct {
 	Settings []PluginSetting
 }
 
+// Session represents a user session.
+type Session struct {
+	ID        uuid.UUID `json:"id"`        // Unique session uuid
+	UserID    uuid.UUID `json:"userid"`    // The user's uuid
+	CreatedAt int64     `json:"createdat"` // Time session was created at
+	MaxAge    int64     `json:"maxage"`    // Max session duration in seconds
+}
+
+func (s Session) HasExpired() bool {
+	// gorm's CreatedAt also uses time.Now()
+	timeNow := time.Now().Unix()
+	return timeNow > (s.CreatedAt + s.MaxAge)
+}
+
+// EncodeSession encodes Session into a JSON byte slice.
+func EncodeSession(s Session) ([]byte, error) {
+	b, err := json.Marshal(s)
+	if err != nil {
+		return nil, err
+	}
+
+	return b, nil
+}
+
+// DecodeSession decodes a JSON byte slice into a Session.
+func DecodeSession(payload []byte) (*Session, error) {
+	var s Session
+
+	err := json.Unmarshal(payload, &s)
+	if err != nil {
+		return nil, err
+	}
+
+	return &s, nil
+}
+
 // Database describes the interface used for interacting with the user
 // database.
 type Database interface {
```

Comment:
Would prefer all the session methods be grouped together.  It makes them easier to grok.

Reply:
This comment still needs to be addressed.

Reply:
Done.